### PR TITLE
Use scale to calculate rect for collision detection

### PR DESCRIPTION
**Pull request by aranasaurus**
Implements #18

### CODE REVIEW

**Comment by aranasaurus** [comment]
Should we also trigger collision detections when an Actor's `scale` changes?

**Comment by JohnSundell** [comment]
@aranasaurus That sounds like a good idea! You want to do it as part of this change as well? You can just call the `rectDidChange()` method in a `didSet {}` on `scale` in Actor, and collision detection will be re-triggered 👍

**Comment by aranasaurus** [comment]
Oh nice, I wasn't sure, so I wanted to check first. I'll do that and add some to the tests for it.

File: Sources/Core/API/Actor.swift [hunk 1]
```diff
@@ -45,7 +45,12 @@ public final class Actor: InstanceHashable, ActionPerformer, Activatable, Movabl
     /// The rotation of the actor along the z axis.
     public var rotation = Metric() { didSet { layer.rotation = rotation } }
     /// The scale of the actor. Does not affect its size, rect or collision detection.
-    public var scale: Metric = 1 { didSet { layer.scale = scale } }
+    public var scale: Metric = 1 {
+        didSet {
+            layer.scale = scale
```
Comment (JohnSundell) [inline Sources/Core/API/Actor.swift]: Could you use the same pattern here as for `position` and `size`, that is put it on a single line (as to not push down the implementation too much), and to delegate to a `scaleDidChange(from:)` method, where you first check if the scale indeed changed and then do the layer update + `rectDidChange()` call? Sorry for the nitpicky details 😅

File: Tests/ImagineEngineTests/ActorTests.swift [hunk 2]
```diff
@@ -298,6 +298,28 @@ final class ActorTests: XCTestCase {
...
+        // A scaled up actor should trigger a collision from farther away.
+        actor.position = Point(x: 300, y: 300)
+        actor.scale = 1.5
+        actor.position = Point(x: 120, y: 120)
+        XCTAssertEqual(numberOfCollisions, 5)
+
+        // Setting a hitbox should override the scaling effect.
+        actor.hitboxSize = Size(width: 25, height: 25)
+        actor.position = Point(x: 90, y: 90)
+        XCTAssertEqual(numberOfCollisions, 5)
```
Comment (JohnSundell) [inline Tests/ImagineEngineTests/ActorTests.swift]: Very nice tests, think this should cover all cases 👍

File: Sources/Core/API/Actor.swift (hunk 1 above)
Comment (aranasaurus) [reply Sources/Core/API/Actor.swift]: ah yeah, that sounds great, not nitpicky at all.

**Comment by aranasaurus** [comment]
@JohnSundell No problem! It was a pleasure. Thank you for setting everything up in a way that made it so easy to jump in and contribute!

**Pull request merged**